### PR TITLE
Validate config file on test

### LINKED ISSUE #391: Add hardhat config file checking
While integrating scarb, I used this in the run_test function of test.sh:
```bash
    npx hardhat --typecheck --config "$CONFIG_FILE_NAME" 1>/dev/null ||
        echo "Invalid config file!"
```

We need a better version of this:
- exit after reporting an invalid config file
- restore the state as it was before the test (or don't change it at all - I was changing it by replacing the config file because I couldn't find another way of validating it).

### PR DESCRIPTION
**Pull request by Nathan-SL**
## Usage related changes

-   Closes #391

## Development related changes

-   Validates config file on test

## Checklist:

-   [x] Formatted the code
-   [x] No linter errors + tried to avoid introducing linter warnings
-   [x] Performed a self-review of the code
-   [x] Rebased to the last commit of the target branch (or merged it into my branch)
-   [ ] Documented the changes
-   [x] Linked issues which this PR resolves
-   [x] All tests are passing (for external contributors who don't have access to the CI/CD pipeline)

### CODE REVIEW

File: scripts/test.sh [hunk 1]
```diff
@@ -53,7 +53,15 @@ function run_test() {
...
     /bin/cp "$config_file_path" "$CONFIG_FILE_NAME"
-    # in the future - validate config file here
+    # Validate config file
+    if npx hardhat --typecheck --config "$CONFIG_FILE_NAME" 1>/dev/null; then
+        echo "Config file is valid"
+    else
+        echo "Invalid config"
+        # Clean up the temporary config file
+        rm -f "$CONFIG_FILE_NAME"
+        exit 1
```
Comment (FabijanC) [inline scripts/test.sh:63]: exit or return? Won't exit cause the whole script to stop, but we want to run all the test?

File: scripts/test.sh [hunk 2]
```diff
@@ -53,7 +53,15 @@ function run_test() {
...
     # replace the dummy config (CONFIG_FILE_NAME) with the one used by this test
     /bin/cp "$config_file_path" "$CONFIG_FILE_NAME"
-    # in the future - validate config file here
+    # Validate config file
+    if npx hardhat --typecheck --config "$CONFIG_FILE_NAME" 1>/dev/null; then
+        echo "Config file is valid"
+    else
+        echo "Invalid config"
+        # Clean up the temporary config file
+        rm -f "$CONFIG_FILE_NAME"
```
Comment (FabijanC) [inline scripts/test.sh:62]: Why `-f`?

**Review by FabijanC — CHANGES_REQUESTED** [review]
Are we sure that tests are not affecting each other? I.e. if one test's config file is not valid, will this have effect no the following test case?

File: scripts/test.sh (hunk 2 above)
Comment (Nathan-SL) [reply scripts/test.sh]: [c2077b6](https://github.com/0xSpaceShard/starknet-hardhat-plugin/pull/401/commits/c2077b6ebd6df6dd642cde161b03518fe5dd8246)

File: scripts/test.sh (hunk 1 above)
Comment (Nathan-SL) [reply scripts/test.sh]: [c2077b6](https://github.com/0xSpaceShard/starknet-hardhat-plugin/pull/401/commits/c2077b6ebd6df6dd642cde161b03518fe5dd8246)

**Review by FabijanC — APPROVED** [review]

**Pull request merged**